javascript/ql/src/semmle/javascript/dataflow/TypeInference.qll

```diff
@@ -47,6 +47,12 @@ class AnalyzedNode extends DataFlow::Node {
    */
   AnalyzedNode localFlowPred() { result = getAPredecessor() }
 
+  /**
+   * Gets another data flow node whose value flows into this node in a global step
+   * (this is, involving global variables).
+   */
+  AnalyzedNode globalFlowPred() { none() }
```

Comment:
I think `global` is too specific. 
Can we call this `nonLocalFlowPred` instead? I think there are many other special cases where it would be interesting to (experimentally) squeeze a bit more out of the type inference.

Reply:
Since it's marked as internal, how about we just rename it when it becomes relevant, rather than trying to predicate its possible future use cases now?